### ISSUE TITLE
outputToTrace has a bad type

### ISSUE THREAD

**Issue by isovector** [polysemy-research/polysemy#286]
`outputToTrace :: (Show w, Member Trace r) => Sem (Output w ': r) a -> Sem r a `

but should be

`outputToTrace :: Member Trace r => (w -> String) => Sem (Output w ': r) a -> Sem r a `

**Comment by juanpaucar** [comment]
Hi @isovector, I'd love to take this ticket. But,  I'm just getting familiar with the library so I might have a PR for the weekend. Would that be okay with you? If so, could I get this issue assigned to me?

**Comment by isovector** [comment]
Hi @juanpaucar --- feel free to tackle this, thanks! No rush on it :)

**Comment by juanpaucar** [comment]
@isovector I just wanted to confirm that the desired function signature is

```hs
outputToTrace :: Member Trace r => (w -> String) -> Sem (Output w ': r) a -> Sem r a

```

Replacing the second `=>` with a `->` on the signature you posted above.

Disclaimer: polisemy is so interesting and I've seen so many new things by reading the code that I don't know what is valid Haskell anymore :sweat_smile:

**Comment by KingoftheHomeless** [comment]
Addressed by #289.

**Comment by isovector** [comment]
Thanks for this, @juanpaucar !
